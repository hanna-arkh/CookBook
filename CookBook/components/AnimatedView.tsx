import React, { useCallback } from 'react'
import Animated, { useSharedValue, withTiming, useAnimatedStyle } from 'react-native-reanimated'
import { useFocusEffect } from '@react-navigation/native'
type AnimatedProps = {
  children: React.ReactNode
}

export const AnimatedView = ({ children }: AnimatedProps) => {
  const opacity = useSharedValue(0)
  useFocusEffect(
    useCallback(() => {
      opacity.value = withTiming(1, { duration: 400 })

      return () => {
        opacity.value = withTiming(0, { duration: 300 })
      }
    }, [])
  )
  const styles = useAnimatedStyle(() => ({
    opacity: opacity.value,
  }))

  return <Animated.View style={[{ flex: 1 }, styles]}>{children}</Animated.View>
}
