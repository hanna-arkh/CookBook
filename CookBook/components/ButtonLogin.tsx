import React from 'react'
import { TouchableOpacity, Text, StyleSheet } from 'react-native'
import { ButtonRegisterProps } from '@/types/types'
import { useTranslation } from 'react-i18next'

export const ButtonLogin = ({ isLoading, disabled, onPress }: ButtonRegisterProps) => {
  const { t } = useTranslation()

  return (
    <TouchableOpacity style={styles.button} onPress={onPress} disabled={disabled || isLoading}>
      <Text>{t('login.loginButton')}</Text>
    </TouchableOpacity>
  )
}
const styles = StyleSheet.create({
  button: {
    backgroundColor: '#007AFF',
    borderRadius: 8,
    paddingVertical: 14,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 40,
  },
})
