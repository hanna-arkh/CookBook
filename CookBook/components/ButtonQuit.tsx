import React from 'react'
import { TouchableOpacity, Text, StyleSheet } from 'react-native'

export const ButtonQuit = () => {
  return (
    <TouchableOpacity style={styles.button}>
      <Text>Quit</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: 'red',
    borderRadius: 8,
    paddingVertical: 7,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 20,
  },
})
