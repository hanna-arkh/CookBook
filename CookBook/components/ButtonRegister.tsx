import React from 'react'
import { TouchableOpacity, Text, StyleSheet } from 'react-native'
import { COLORS, LAYOUT } from '@/constants/Constants'
import { ButtonRegisterProps } from '@/types/types'
import { useTranslation } from 'react-i18next'

export const ButtonRegister = ({ onPress, isLoading, disabled }: ButtonRegisterProps) => {
  const { t } = useTranslation()

  return (
    <TouchableOpacity
      style={[styles.button, isLoading && styles.disabledButton]}
      onPress={onPress}
      disabled={disabled || isLoading}
    >
      <Text>{t('login.signUp')}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: COLORS.BUTTON_REGISTRATION,
    borderRadius: 8,
    paddingVertical: 14,
    alignItems: LAYOUT.ALIGN.CENTER,
    justifyContent: LAYOUT.ALIGN.CENTER,
    paddingHorizontal: 40,
  },
  disabledButton: {
    opacity: 0.5,
  },
})
