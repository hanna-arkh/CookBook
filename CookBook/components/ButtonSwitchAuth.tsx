import React from 'react'
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
import { useTranslation } from 'react-i18next'
import { ROUTES, COLORS } from '@/constants/Constants'
import { useRouter, useLocalSearchParams } from 'expo-router'

export default function ButtonSwitchAuth() {
  const { t } = useTranslation()
  const router = useRouter()
  const { mode } = useLocalSearchParams<{ mode: 'register' | 'index ' }>()
  const isRegister = mode === 'register'
  const routeName = isRegister ? ROUTES.AUTH : ROUTES.REGISTER
  const linkText = isRegister ? t('login.loginButton') : t('login.signUp')

  return (
    <View>
      <TouchableOpacity onPress={() => router.navigate(routeName)}>
        <Text style={styles.link}>{linkText}</Text>
      </TouchableOpacity>
    </View>
  )
}
const styles = StyleSheet.create({
  link: {
    marginTop: 20,
    color: COLORS.LINK_COLOR,
  },
})
