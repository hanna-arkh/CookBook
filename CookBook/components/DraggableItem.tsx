import React from 'react'
import { Text, Image, StyleSheet, View } from 'react-native'
import { Recipe } from '@/types/types'
import { COLORS } from '@/constants/Colors'
import { Gesture, GestureDetector } from 'react-native-gesture-handler'
import Animated, { useSharedValue, useAnimatedStyle, withSpring } from 'react-native-reanimated'

const AnimatedView = Animated.createAnimatedComponent(View)

export default function DraggableItem({ item }: { item: Recipe }) {
  const offsetX = useSharedValue(0)
  const offsetY = useSharedValue(0)

  const panGesture = Gesture.Pan()
    .onUpdate(event => {
      offsetX.value = event.translationX
      offsetY.value = event.translationY
    })
    .onEnd(() => {
      offsetX.value = withSpring(0)
      offsetY.value = withSpring(0)
    })

  const animatedStyle = useAnimatedStyle(() => ({
    transform: [{ translateX: offsetX.value }, { translateY: offsetY.value }],
  }))

  return (
    <GestureDetector gesture={panGesture}>
      <AnimatedView style={[recipeCardStyles.card, animatedStyle]}>
        <Image source={{ uri: item.strMealThumb }} style={recipeCardStyles.image} />
        <Text style={recipeCardStyles.text}>{item.strMeal}</Text>
      </AnimatedView>
    </GestureDetector>
  )
}

const recipeCardStyles = StyleSheet.create({
  card: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 10,
    marginVertical: 8,
    marginHorizontal: 16,
    backgroundColor: COLORS.LIST_ITEM_BG,
    borderRadius: 10,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  image: {
    width: 60,
    height: 60,
    borderRadius: 30,
    marginRight: 10,
  },
  text: {
    fontSize: 16,
  },
})
