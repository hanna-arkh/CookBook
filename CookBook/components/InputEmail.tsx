import React from 'react'
import { TextInput, View, StyleSheet } from 'react-native'
import { COLORS, LAYOUT, AUTH, UI_LABELS } from '@/constants/Constants'
import { PropsEmail } from '@/types/types'
export const InputEmail = ({ value, onChangeText }: PropsEmail) => {
  return (
    <View style={styles.container}>
      <TextInput
        style={styles.input}
        placeholder={UI_LABELS.EMAIL}
        keyboardType={AUTH.EMAIL_KEYBOARD}
        autoCorrect={false}
        value={value}
        onChangeText={onChangeText}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    width: LAYOUT.WIDTH.FULL,
    marginBottom: 16,
  },
  input: {
    backgroundColor: COLORS.INPUT_BG_COLOR,
    borderWidth: 1,
    borderColor: COLORS.INPUT_BORDER,
    borderRadius: 8,
    paddingHorizontal: 16,
    paddingVertical: 12,
    fontSize: 16,
    color: COLORS.BLACK,
    marginHorizontal: 20,
  },
})
