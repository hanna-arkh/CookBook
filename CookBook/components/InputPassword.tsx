import React from 'react'
import { TextInput, View, StyleSheet } from 'react-native'
import { COLORS, LAYOUT, AUTH } from '@/constants/Constants'
import { PropsPassword } from '@/types/types'
import { useTranslation } from 'react-i18next'

export const InputPassword = ({ value, onChangeText }: PropsPassword) => {
  const { t } = useTranslation()

  return (
    <View style={styles.container}>
      <TextInput
        style={styles.input}
        placeholder={t('login.password')}
        placeholderTextColor={COLORS.GREY}
        keyboardType={AUTH.EMAIL_KEYBOARD}
        autoCorrect={false}
        secureTextEntry
        value={value}
        onChangeText={onChangeText}
      />
    </View>
  )
}
const styles = StyleSheet.create({
  container: {
    width: LAYOUT.WIDTH.FULL,
    marginBottom: 16,
  },
  input: {
    backgroundColor: COLORS.INPUT_BG_COLOR,
    borderWidth: 1,
    borderColor: COLORS.INPUT_BORDER,
    borderRadius: 8,
    paddingHorizontal: 16,
    paddingVertical: 12,
    fontSize: 16,
    color: COLORS.BLACK,
    marginHorizontal: 20,
  },
})
