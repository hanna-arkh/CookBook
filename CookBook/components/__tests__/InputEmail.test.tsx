import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import { InputEmail } from '@/components/InputEmail'
import i18n from '@/services/i18n'
describe('InputEmail Component', () => {
  test('renders email input with placeholder in english', () => {
    const { getByPlaceholderText } = render(<InputEmail value="" onChangeText={() => {}} />)
    expect(getByPlaceholderText(i18n.t('login.email'))).toBeTruthy()
  })

  test('shows correct email value', () => {
    const email = 'user@test.com'
    const { getByDisplayValue } = render(<InputEmail value={email} onChangeText={() => {}} />)
    expect(getByDisplayValue(email)).toBeTruthy()
  })
  test('calls onChange when typing in enlish', () => {
    const onChangeMock = jest.fn()
    const { getByPlaceholderText } = render(<InputEmail value="" onChangeText={onChangeMock} />)
    fireEvent.changeText(getByPlaceholderText(i18n.t('login.email')), 'test@mail.com')
    expect(onChangeMock).toHaveBeenCalledWith('test@mail.com')
  })

  test('looks correct (snapshot)', () => {
    const { toJSON } = render(<InputEmail value="test@example.com" onChangeText={() => {}} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
