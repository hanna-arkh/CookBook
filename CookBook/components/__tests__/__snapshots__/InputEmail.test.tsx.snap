// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputEmail Component looks correct (snapshot) 1`] = `
<View
  style={
    {
      "marginBottom": 16,
      "width": "100%",
    }
  }
>
  <TextInput
    autoCorrect={false}
    keyboardType="email-address"
    onChangeText={[Function]}
    placeholder="Email"
    style={
      {
        "backgroundColor": "#f0f0f0",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "color": "black",
        "fontSize": 16,
        "marginHorizontal": 20,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      }
    }
    value="test@example.com"
  />
</View>
`;
